lib/internal/modules/esm/get_format.js

```diff
@@ -154,15 +163,14 @@ function getFileProtocolModuleFormat(url, context = { __proto__: null }, ignoreE
         return 'commonjs';
       }
       default: { // The user did not pass `--experimental-default-type`.
-        if (getOptionValue('--experimental-detect-module')) {
-          if (!source) { return null; }
-          const format = getFormatOfExtensionlessFile(url);
-          if (format === 'module') {
-            return containsModuleSyntax(`${source}`, fileURLToPath(url), url) ? 'module' : 'commonjs';
-          }
+        if (!source) {
+          return;
```

Comment:
Why not keep `return null`?
```suggestion
          return null;
```

Reply:
Why should users need to check another potential value? I don't see what it adds. Better a string or undefined, rather than a string or null or undefined.

Reply:
Maybe don’t mark conversations as resolved if they aren’t resolved.
They’ll have to check anyway, because other hooks may set it to `null`. It previously returned `null`, changing it seems to only makes things more complicated

Reply:
We went through this with https://github.com/nodejs/node/pull/53015 and associated PRs. I don’t really want to open the can of worms again that is `undefined` vs `null`. There’s no benefit that I can see to it sometimes returning `null`.

Reply:
Im sorry, but we don’t make a breaking change to our users just without a good reason, and a dedicated commit so it appears in our CHANGELOG. This change is unrelated to this PR, it should be its own PR – which as you pointed out is #53015 already. If you don’t want to open that can of worms, don’t open it sneakily in an unrelated PR

Reply:
There's nothing sneaky. This PR changes the return value from `commonjs` to undefined. Just look at the changed tests. That's what unflagging does, and what I wrote about at length in the top post.